### PR TITLE
Play bundle plugin

### PR DESCRIPTION
**Pull request by markusjura**
Adds the bundle plugin for Play projects. This auto plugin get triggered for each `PlayJava` or `PlayScala` project. This plugin, compared to the normal bundle plugin, adds default bundle key values and the play conductr bundle library to the Play project.

### CODE REVIEW

**Comment by markusjura** [comment]
Now `sbt-conductr` does not include the Lagom and Play sbt plugins as a dependency. Having them as dependencies resulted into version issues. If `sbt-conductr` added Play 2.5.x then it was not usable by Play 2.3.x and vice versa.

Instead we are using reflection to check if the particular plugin (e.g. Play or LagomJava) is part of the classpath. The same logic has been applied to the `LagomBundlePlugin`. Tests for Play 2.3, 2.4 and 2.5 have been added.

Also I've created a package object for functions or values that are used across different plugins.

File: src/main/scala/com/lightbend/conductr/sbt/LagomBundlePlugin.scala [hunk 1]
```diff
@@ -33,19 +38,19 @@ object LagomBundlePlugin extends AutoPlugin {
   override def projectSettings =
     bundleSettings(Bundle) ++ Seq(
       BundleKeys.bundleConfVersion := BundleConfVersions.V_1_2_0,
-      BundleKeys.nrOfCpus := 1.0,
-      BundleKeys.memory := 128.MiB,
-      BundleKeys.diskSpace := 200.MB,
+      BundleKeys.nrOfCpus := PlayBundleKeyDefaults.nrOfCpus,
+      BundleKeys.memory := PlayBundleKeyDefaults.memory,
+      BundleKeys.diskSpace := PlayBundleKeyDefaults.diskSpace,
```
Comment (huntc) [inline src/main/scala/com/lightbend/conductr/sbt/LagomBundlePlugin.scala]: Nice

File: src/main/scala/com/lightbend/conductr/sbt/LagomBundlePlugin.scala [hunk 2]
```diff
@@ -20,11 +17,19 @@ object LagomBundlePlugin extends AutoPlugin {
   import LagomBundleKeys._
   import SbtNativePackager.autoImport._
   import BundlePlugin.autoImport._
-  import ByteConversions._
 
   val autoImport = LagomBundleImport
 
-  override def requires = LagomJava && BundlePlugin
+  private val classLoader = this.getClass.getClassLoader
```
Comment (huntc) [inline src/main/scala/com/lightbend/conductr/sbt/LagomBundlePlugin.scala]: `pluginClassLoader` may be a better name - more descriptive.

File: src/main/scala/com/lightbend/conductr/sbt/LagomBundlePlugin.scala [hunk 3]
```diff
@@ -274,37 +285,59 @@ private object ServiceDetector {
...
+/**
+ * Mirrors the LagomVersion class of `com.lightbend.lagom.sbt.LagomImport`
+ * By declaring the public methods from Lagom it is possible to "safely"
+ * call the class via reflection.
+ */
+private object LagomImport {
+
+  import scala.language.reflectiveCalls
+
+  val classLoader = this.getClass.getClassLoader
```
Comment (huntc) [inline src/main/scala/com/lightbend/conductr/sbt/LagomBundlePlugin.scala]: `pluginsClassLoader`

File: src/main/scala/com/lightbend/conductr/sbt/PlayBundleImport.scala [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.lightbend.conductr.sbt
+
+import sbt._
+
+object PlayBundleImport {
+
+  val conductrBundleLibVersion = BaseKeys.conductrBundleLibVersion
+
+  val playClassLoading = taskKey[String]("play-class-loading")
```
Comment (huntc) [inline src/main/scala/com/lightbend/conductr/sbt/PlayBundleImport.scala]: What does this do? It isn't namespaced either...

Comment (huntc) [inline src/main/scala/com/lightbend/conductr/sbt/PlayBundleImport.scala]: Should at least be commented, and possibly made private.

File: src/main/scala/com/lightbend/conductr/sbt/package.scala [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+  final val Sha256 = "SHA-256"
+  final val Utf8 = Charset.forName("utf-8")
+
+  /**
+   * Create a hash based on a UTF-8 string
+   */
+  def hash(content: String): String =
+    hash(content.getBytes(Utf8))
+
+  def hash(bytes: Array[Byte]): String =
```
Comment (huntc) [inline src/main/scala/com/lightbend/conductr/sbt/package.scala]: All public functions should receive scala doc.

File: src/main/scala/com/lightbend/conductr/sbt/package.scala [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    val conductrBundleLibVersion = SettingKey[String](
+      "play-bundle-conductr-bundle-lib-version",
+      "The version of conductr-bundle-lib to depend on. Defaults to 1.4.2"
+    )
+  }
+
+  /**
+   * Default bundle keys for a Play project
+   */
+  object PlayBundleKeyDefaults {
```
Comment (huntc) [inline src/main/scala/com/lightbend/conductr/sbt/package.scala]: Should this be in a package object? It isn't applicable to all...

Comment (huntc) [inline src/main/scala/com/lightbend/conductr/sbt/package.scala]: Why don't we see similar constants for Lagom, or was that another PR?

**Pull request merged**